Kernel/Syscalls/uname.cpp

```diff
@@ -13,19 +16,67 @@ ErrorOr<FlatPtr> Process::sys$uname(Userspace<utsname*> user_buf)
     VERIFY_NO_PROCESS_BIG_LOCK(this);
     TRY(require_promise(Pledge::stdio));
 
-    utsname buf {};
-    memcpy(buf.sysname, "SerenityOS", 11);
-    memcpy(buf.release, "1.0-dev", 8);
-    memcpy(buf.version, "FIXME", 6);
+    utsname buf
+    {
+        "SerenityOS",
+            "", // Hostname, filled in below.
+            "", // "Release" (1.0-dev), filled in below.
+            "", // "Revision" (git commit hash), filled in below.
 #if ARCH(I386)
-    memcpy(buf.machine, "i686", 5);
+            "i686",
 #elif ARCH(X86_64)
-    memcpy(buf.machine, "x86_64", 7);
+            "x86_64",
 #elif ARCH(AARCH64)
-    memcpy(buf.machine, "AArch64", 7);
+            "AArch64",
 #else
 #    error Unknown architecture
 #endif
+    };
+
+    // The full version is found in /res/version.ini, so let's read that like the user would.
+    auto description = TRY(VirtualFileSystem::the().open(credentials(), "/res/version.ini"sv, O_RDONLY, 0000, current_directory()));
+    ScopeGuard guard([&]() { (void)description->close(); });
+    auto raw_contents = TRY(description->read_entire_file());
+    auto contents = StringView(raw_contents->bytes());
+
+    // FIXME: This INI parsing is super crude but we can't use Core::ConfigFile. As long as our version.ini doesn't do weird things, it should work.
+    // Find a way of reusing code across Kernel and Userland.
+    auto maybe_section_location = contents.find("[Version]\n"sv);
+    if (maybe_section_location.has_value()) {
+        auto section_location = maybe_section_location.value() + 10;
+        auto major_version_location = contents.find("Major="sv, section_location);
+        auto minor_version_location = contents.find("Minor="sv, section_location);
+        auto git_hash_location = contents.find("Git="sv, section_location);
```

Comment:
I hadn't even seen this, now I dislike the idea of parsing version.ini in the kernel even more :sweat_smile:

Reply:
Let's see how the #15071 discussion goes.